src/commcare_cloud/ansible/library/setup_virtualenv.py

```diff
@@ -115,30 +113,54 @@ def main():
     if not python_env.exists():
         result["changed"] = True
         if not module.check_mode:
-            if not (next_env / "bin/python").exists():
-                if not prev_env.exists():
-                    module.fail_json(msg=f"virtualenv not found: {prev_env}")
-                    return
-                clone_virtualenv(prev_env, next_env, module)
-            pip_sync(requirements_file, next_env, module, proxy)
+            if (dest / "pyproject.toml").exists():
+                uv_sync(dest, proxy, module)
+                assert (dest / ".venv").is_dir(), "uv did not create .venv"
+                (dest / full_env_name).symlink_to(".venv")
+                full_env_name = ".venv"
+            else:
+                if not (next_env / "bin/python").exists():
+                    if not prev_env.exists():
+                        module.fail_json(msg=f"virtualenv not found: {prev_env}")
+                        return
+                    clone_virtualenv(prev_env, next_env, module)
+                pip_sync(dest, next_env, module, proxy)
             python_env.symlink_to(full_env_name)
 
     module.exit_json(**result)
 
 
+def uv_sync(dest, proxy, module):
+    proxy_env = {"ALL_PROXY": proxy} if proxy else {}
+    module.run_command(
+        ["uv", "sync", "--group=prod", "--no-dev", "--locked", "--compile-bytecode"],
```

Comment:
How do you feel about adding a comment explaining why we use compile-bytecode and when we would be able to remove it? The other parameters are intuitive.

Reply:
I think we should keep it indefinitely (regardless of weird issues it fixed). It speeds up first-time interpreter startup.